diff/diff.go

```diff
@@ -0,0 +1,293 @@
+package diff
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"reflect"
+	"sort"
+	"strings"
+)
+
+// Difference represents a "chunk" of difference between two
+// `Object`s.
+type Difference interface {
+	Summarise(out io.Writer)
+	AtPath() string
```

Comment:
Does a difference always have a path? 
E.g. if there is no file in the repository then there won't be a path, right?